### PR TITLE
eth/downloader: prioritize block fetch based on chain position, cap memo...

### PR DESCRIPTION
**Pull request by karalabe**
I've reworked the downloader queue to prioritize based on the blockchain position, so that you don't need to pool everything to import, but rather can do it concurrently. In addition, I've places a 4096 block download cap, after which fetches should be throttled until the processor catches up. There might be some bugs in this cap!

Also cleaned up the downloader to separate all queue related stuff into the queue.go file.

@obscuren If you have time, just look through it as a discussion starter.

### CODE REVIEW

**Comment by obscuren** [comment]
Tried downloading and blocks at some point.

**Comment by karalabe** [comment]
Ah, crap. There's probably some weird corner case in the memory cap. Will
try to look into it tonigjt.

File: eth/sync.go [hunk 1]
```diff
@@ -78,26 +77,19 @@ func (pm *ProtocolManager) processBlocks() error {
 	return nil
 }
 
-func (pm *ProtocolManager) synchronise(peer *peer) {
+func (pm *ProtocolManager) synchronize(peer *peer) {
```
Comment (obscuren) [inline eth/sync.go]: Stick with proper english `synchronise`

Comment (karalabe) [inline eth/sync.go]: k :)

**Comment by karalabe** [comment]
PTAL

**Pull request merged**